lib/galaxy/datatypes/sniff.py

```diff
@@ -198,7 +198,12 @@ def get_headers( fname, sep, count=60, is_multi_byte=False ):
     [['chr7', '127475281', '127491632', 'NM_000230', '0', '+', '127486022', '127488767', '0', '3', '29,172,3225,', '0,10713,13126,'], ['chr7', '127486011', '127488900', 'D49487', '0', '+', '127486022', '127488767', '0', '2', '155,490,', '0,2399']]
     """
     headers = []
-    for idx, line in enumerate(open(fname)):
+    compressed = is_gzip(fname)
+    if compressed:
+        in_file = gzip.GzipFile(fname, 'r')
+    else:
+        in_file = open(fname, 'rt')
```

Comment:
quickly scanning through the datatypes, some functions do and some don't. I'll add this.